appinventor/components/src/com/google/appinventor/components/runtime/util/ViewUtil.java

```diff
@@ -189,9 +198,57 @@ public static void setBackgroundImage(View view, Drawable drawable) {
   /**
    * Sets the image for an ImageView.
    */
-  public static void setImage(ImageView view, Drawable drawable) {
-    view.setImageDrawable(drawable);
-    if (drawable != null) {
+  public static void setImage(ComponentContainer container, ImageView view, String picturePath) {
+    if (picturePath.substring(picturePath.lastIndexOf('.') + 1).equals("gif")) {
```

Comment:
Rather than use lastIndexOf, I would recommend using `picturePath.endsWith()` on the off chance that someone just includes an asset called "gif".

Second, there should be an additional check here that this code path is only taken on SDK 10 and higher.

Reply:
@ewpatton Made the changes. Kindly review.